From 3be7cda6e0b58051fcf761a694fd6065dc9d84a1 Mon Sep 17 00:00:00 2001
From: Charles Bousseau <cbousseau@anaconda.com>
Date: Wed, 14 Jun 2023 22:52:16 -0400
Subject: [PATCH 01/22] branding

---
 Lib/platform.py     | 2 +-
 Python/getversion.c | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Lib/platform.py b/Lib/platform.py
index 8895177e326..df25c2c04b4 100755
--- a/Lib/platform.py
+++ b/Lib/platform.py
@@ -1191,7 +1191,7 @@ def _sys_version(sys_version=None):
     else:
         # CPython
         cpython_sys_version_parser = re.compile(
-            r'([\w.+]+)\s*'  # "version<space>"
+            r'([\w.+]+)\s*(?:\ \|\ packaged\ by\ Anaconda,\ Inc\.\ \|)?\s*'  # version<space>"
             r'(?:experimental free-threading build\s+)?' # "free-threading-build<space>"
             r'\(#?([^,]+)'  # "(#buildno"
             r'(?:,\s*([\w ]*)'  # ", builddate"
diff --git a/Python/getversion.c b/Python/getversion.c
index 226b2f999a6..0c7ff163257 100644
--- a/Python/getversion.c
+++ b/Python/getversion.c
@@ -6,7 +6,7 @@
 #include "patchlevel.h"
 
 static int initialized = 0;
-static char version[300];
+static char version[350];
 
 void _Py_InitVersion(void)
 {
@@ -15,9 +15,9 @@ void _Py_InitVersion(void)
     }
     initialized = 1;
 #ifdef Py_GIL_DISABLED
-    const char *buildinfo_format = "%.80s experimental free-threading build (%.80s) %.80s";
+    const char *buildinfo_format = "%.80s | packaged by Anaconda, Inc. | experimental free-threading build (%.80s) %.80s";
 #else
-    const char *buildinfo_format = "%.80s (%.80s) %.80s";
+    const char *buildinfo_format = "%.80s | packaged by Anaconda, Inc. | (%.80s) %.80s";
 #endif
     PyOS_snprintf(version, sizeof(version), buildinfo_format,
                   PY_VERSION, Py_GetBuildInfo(), Py_GetCompiler());
